clippy_lints/src/types.rs

```diff
@@ -171,11 +171,38 @@ declare_clippy_lint! {
     "a borrow of a boxed type"
 }
 
+declare_clippy_lint! {
+    /// **What it does:** Checks for use of `Box<&T>` anywhere in the code.
+    ///
+    /// **Why is this bad?** Any `Box<&T>` can also be a `&T`, which is more
+    /// general.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    /// ```rust,ignore
+    /// struct X {
+    ///     values: Box<&T>,
+    /// }
+    /// ```
+    ///
+    /// Better:
+    ///
+    /// ```rust,ignore
+    /// struct X {
+    ///     values: &T,
+    /// }
+    /// ```
+    pub BOX_BORROW,
```

Comment:
```suggestion
    pub BOX_BORROWS,
```

This will require to also update the names in the rest of the file , plus updating the tests and running `cargo dev update_lints`